eng/pipelines/templates/pipeline.yml

```diff
@@ -74,6 +78,7 @@ stages:
   jobs:
   - job: RunStaticSourceAnalysis
     displayName: Run Static Source Analysis
+    condition: "and(succeeded(), ne(variables['RunStaticAnalysis'], 'false'))"
```

Comment:
Did you test this?  As far as I know, parameters != variables, so I don't think this will work.

Reply:
From https://learn.microsoft.com/en-us/azure/devops/pipelines/process/runtime-parameters?view=azure-devops&tabs=script

> Parameters are only available at template parsing time. Parameters are expanded just before the pipeline runs so that values surrounded by ${{ }} are replaced with parameter values. Use [variables](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops) if you need your values to be more widely available during your [pipeline run](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/runs?view=azure-devops).

Reply:
Yes it works but only because we copy some parameters to variables: https://github.com/NuGet/NuGet.Client/blob/dev-jeffkl-move-static-analysis/eng/pipelines/pull_request.yml#L65

Reply:
![image](https://github.com/NuGet/NuGet.Client/assets/17556515/c2cc1d70-a913-419a-bc85-d9e2e1b4deec)

Reply:
Ah, in that case there's no benefit to adding the `RunStaticAnalysis` parameter to this pipeline.yml file, since it's an unused parameter.

Reply:
I'm not sure, but does `condition: "and(succeeded(), ne( '${{ parameter.RunStaticAnalysis }}', 'false')"` work?

It might be a way to avoid needing to declare a bunch of variables that are only used once?

Reply:
At the moment I'm not too worried, I'm hoping in the near future to clean up PrivateDev and have it just build and get rid off all these other stages, jobs, parameters, and jobs.  My primary concern is leaving things as-is so that if something is wrong we can bring back previous behavior.